### PR TITLE
[+] add `pgx.QueryRewriter` support

### PR DESCRIPTION
**Pull request by pashagolub**

### CODE REVIEW

File: expectations.go [hunk 1]
```diff
@@ -130,28 +130,58 @@ func (e *commonExpectation) String() string {
...
+func (e *queryBasedExpectation) argsMatches(sql string, args []interface{}) (rewrittenSQL string, err error) {
+	eargs := e.args
+	// check for any query rewriters
+	// according to current pgx docs, a QueryRewriter is only supported as the first
+	// argument.
+	if len(args) == 1 {
+		if qrw, ok := args[0].(pgx.QueryRewriter); ok {
+			// note: pgx.Conn is not currently used by the query rewriter, but is part
+			// of the method signature, so just create an empty pointer for now.
+			rewrittenSQL, newArgs, err := qrw.RewriteQuery(context.Background(), new(pgx.Conn), sql, args)
```
Comment (dropwhile) [inline expectations.go:147]: suggestion (if-minor): Possibly reduce chance for future bugs

#### Detail
Which variables in the [short variable decl](https://go.dev/ref/spec#Short_variable_declarations) are being overwritten and which get masked?
I _think_ this works as expected (rewrittenSQL and err getting set without being masked, while newArgs gets created), but I've found this construction (partial redecl of anything other than err) to be a bit prone to being a future source of bugs.

I might recommend instead:
```suggestion
            var newArgs []any
			rewrittenSQL, newArgs, err = qrw.RewriteQuery(context.Background(), new(pgx.Conn), sql, args)
```

or perhaps:
```suggestion
			rewrittenSQL, args, err = qrw.RewriteQuery(context.Background(), new(pgx.Conn), sql, args)
```

File: pgxmock.go [hunk 2]
```diff
@@ -430,7 +432,15 @@ func (c *pgxmock) Query(ctx context.Context, sql string, args ...interface{}) (p
 		if err := c.queryMatcher.Match(queryExp.expectSQL, sql); err != nil {
 			return err
 		}
-		if err := queryExp.argsMatches(args); err != nil {
+		if rewrittenSQL, err := queryExp.argsMatches(sql, args); err != nil {
+			return err
+		} else if rewrittenSQL > "" {
```
Comment (dropwhile) [inline pgxmock.go:437]: question (non-blocking): curious why `>` was used instead of `!=`?

**Review by dropwhile — COMMENTED** [review]
minor feedback comments in conventionalcomments.org style

File: pgxmock.go (hunk 2 above)
Comment (pashagolub) [reply pgxmock.go:437]: Old habit. In some old languages and compilers this was faster :-)

**Comment by pashagolub** [comment]
TIL

**Review by dropwhile — APPROVED** [review]
looks good to me 👍

**Pull request merged**